server/data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java

```diff
@@ -328,6 +328,25 @@ public void sortMappingsShouldPutDomainAliasFirstWhenVariousMappings() {
                         .build());
     }
 
+    @Test
+    public void addMappingShouldThrowWhenMappingAlreadyExists() throws Exception {
+        String user = "test";
+        String domain = "localhost";
+        String address = "test@localhost2";
+
+        assertThat(addMapping(user, domain, address, ADDRESS_TYPE)).isTrue();
+        assertThat(addMapping(user, domain, address, ADDRESS_TYPE)).isFalse();
```

Comment:
what line is the when and what line it then ?

Reply:
first is when and second is then? :)
no seriously intent would be clearer with exception but see my previous comment